tests/5.0/requires/test_requires_dynamic_allocators.c

```diff
@@ -0,0 +1,64 @@
+//===------ test_requires_dynamic_allocators.c ----------------------------===//
+//
+// OpenMP API Version 5.0 Nov 2018
+//
+// Tests the requires dynamic allocators clause by working with an OMP
+// allocator inside a target region. Allocators testing is based on the
+// OpenMP 5.0 example for allocators. The allocator testing first creates
+// an allocator, with 64-byte alignment and the default memory space,
+// then checks that 64-byte alignment is correct and that the memory can
+// be written to in the target region. The tests checks that the values
+// were written correctly, and then frees the memory and deletes the
+// allocator.
+//
+//===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int test_requires() {
+  int errors = 0;
+
+#pragma omp target map(from: errors)
+  {
+    int* x;
+    omp_memspace_t x_memspace = omp_default_mem_space;
+    omp_alloctrait_t x_traits[1] = {omp_atk_alignment, 64};
+    omp_allocator_handle_t x_alloc = omp_init_allocator(x_memspace, 1, x_traits);
+
+    x = (int *) omp_alloc(N*sizeof(int), x_alloc);
+
+    // Error if x is not 64-byte aligned (address should be divisible by 64)
+    OMPVV_TEST_AND_SET_VERBOSE(errors, ((intptr_t)(x)%64 != 0));
```

Comment:
Based on summit results it is possible Clang and XLC do not define the intptr_t type. I might look into a different way to test this.